vital/algo/metadata_map_io.h

```diff
@@ -0,0 +1,139 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/**
+ * \file
+ * \brief Interface for serialize_metadata \link
+ * kwiver::vital::algo::algorithm_def algorithm
+ *        definition \endlink.
+ */
+
+#ifndef VITAL_ALGO_METADATA_MAP_IO_H_
+#define VITAL_ALGO_METADATA_MAP_IO_H_
+
+#include <vital/algo/algorithm.h>
+
+#include <vital/algorithm_capabilities.h>
+
+#include <vital/types/metadata.h>
+#include <vital/types/metadata_map.h>
+
+#include <vital/vital_config.h>
+
+#include <string>
+
+namespace kwiver {
+
+namespace vital {
+
+namespace algo {
+
+/// An abstract base class for reading and writing metadata maps.
+///
+/// This class represents an abstract interface for reading and writing video
+/// metadata.
+class VITAL_ALGO_EXPORT metadata_map_io
+  : public kwiver::vital::algorithm_def< metadata_map_io >
+{
+public:
+  virtual ~metadata_map_io() = default;
+
+  /// Return the name of this algorithm
+  static std::string static_type_name() { return "metadata_map_io"; }
+
+  /// Load metadata from the file.
+  ///
+  /// \throws kwiver::vital::path_not_exists
+  ///   Thrown when the given path does not exist.
+  /// \throws kwiver::vital::path_not_a_file
+  ///   Thrown when the given path does not point to a file (i.e. it points to
+  ///   a directory).
+  ///
+  /// \param filename The path to the file to load.
+  /// \returns A metadata_map_sptr pointing to the data.
+  kwiver::vital::metadata_map_sptr load( std::string const& filename ) const;
+
+  /// Load metadata from the file.
+  ///
+  /// \param fin The stream to read from.
+  /// \param filename The path to the file to load.
```

Comment:
I'd suggest something closer to the `save` documentation? "Filename that generated the stream." You might also say (in both places) "May be used for logging, or in case of other operations for which the filesystem path to the input is relevant."

Maybe make similar changes for `load_` and `save_`?